### DISCUSSION TITLE
Xwayland support?

### DISCUSSION THREAD

**Discussion by wesbluemarine** [YaLTeR/niri#60] [Q&A] [answered]
Is there any xwayland support?

**Comment by YaLTeR** [comment, marked as answer]
There isn't, and unfortunately due to X11 being completely cursed, built-in Xwayland support is not planned at the moment.

You can, however, use X11 apps on niri using Valve's gamescope (game-oriented), or by running Xwayland in rootful mode:

1. Run `Xwayland` (just the binary on its own without flags). This will spawn a large black monitor-sized window, you can fullscreen it for convenience. In the next Xwayland version, it will be properly resizable, too.
2. Run some X11 window manager in there, e.g. `env DISPLAY=:0 i3`. This way you can manage X11 windows inside the Xwayland instance.
3. Run an X11 application there, e.g. `env DISPLAY=:0 flatpak run com.valvesoftware.Steam`.

**Comment by KiaraGrouwstra** [comment]
i actually managed to successfully use wine through [`xwayland-run`](https://gitlab.freedesktop.org/ofourdan/xwayland-run) like `xwayland-run -- wine game.exe` - perhaps that tool might be worth mentioning over at the wiki article as well. 😅

**Reply by YaLTeR** [reply]
Looks interesting, yeah.

**Reply by salman-farooq-sh** [reply]
I can make a PR for a small write-up on xwayland-run in the wiki if you want. I think this should be at the top of the currently suggested solutions.

**Reply by YaLTeR** [reply]
Sure

**Comment by leiserfg** [comment]
Smithay already includes a XWayland support [implementation](https://github.com/Smithay/smithay/tree/master/src/xwayland), so do [anvil ](https://github.com/Smithay/smithay/blob/master/anvil/src/x11.rs)  and  [Cosmic](https://github.com/pop-os/cosmic-comp/blob/master/src/backend/x11.rs) can this happen in niri?

**Reply by YaLTeR** [reply]
It's challenging to do in niri due to the lack of a global coordinate system for windows, among other things.
